hail_v02_scripts/utils/elasticsearch/client.py

```diff
@@ -0,0 +1,253 @@
+import datetime
+import logging
+import re
+import time
+from pprint import pformat
+
+import elasticsearch
+import hail as hl
+
+from .schema import elasticsearch_schema_for_table
+from .utils import encode_field_name
+
+
+logger = logging.getLogger()
+
+
+# Elastic search write operations.
+# See https://www.elastic.co/guide/en/elasticsearch/hadoop/current/configuration.html#_operation
+ELASTICSEARCH_INDEX = "index"
+ELASTICSEARCH_CREATE = "create"
+ELASTICSEARCH_UPDATE = "update"
+ELASTICSEARCH_UPSERT = "upsert"
+ELASTICSEARCH_WRITE_OPERATIONS = set(
+    [ELASTICSEARCH_INDEX, ELASTICSEARCH_CREATE, ELASTICSEARCH_UPDATE, ELASTICSEARCH_UPSERT]
+)
+
+
+class ElasticsearchClient:
+    def __init__(self, host="localhost", port="9200"):
+        """Constructor.
+
+        Args:
+            host (str): Elasticsearch server host
+            port (str): Elasticsearch server port
+        """
+
+        self._host = host
+        self._port = port
+
+        self.es = elasticsearch.Elasticsearch(host, port=port)
+
+    def create_or_update_mapping(self, index_name, index_type_name, elasticsearch_schema={}, num_shards=1, _meta=None):
```

Comment:
Some code, like this method and the functions to encode/decode field names are duplicated between here and `hail_scripts`. We could organize `hail_scripts` into something like:
```
hail_scripts/
  v01/
  v02/
  shared/
```
if we wanted to avoid the duplication.

Reply:
👍. If that's the way we want to go, I can submit a separate PR to reorganize the existing code before adding this new code. There's quite a few path names and imports to change 😭.

Reply:
Whatever you guys decide to do with @mattsolo1 ..